libs/blockchain/src/blockchain.service.ts

```diff
@@ -89,25 +89,27 @@ export class BlockchainService extends BlockchainRpcQueryService implements OnAp
   }
 
   public async onApplicationShutdown() {
+    this.logger.info('received shutdown event');
```

Comment:
This log could be misleading; it's not necessarily a foregone conclusion that the app is shutting down due to the `'shutdown'` event. Perhaps:
```suggestion
    this.logger.info('Processing application shutdown');
```